### PR TITLE
Create clds_reals lib

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Create clds_reals lib

### CODE REVIEW

File: tests/reals/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    real_clds_sorted_list.h
+    real_clds_sorted_list_renames.h
+    real_clds_st_hash_set.h
+    real_clds_st_hash_set_renames.h
+    real_lock_free_set.h
+    real_lock_free_set_renames.h
+)
+
+include_directories(${CMAKE_CURRENT_LIST_DIR}/../../src)
+add_library(clds_reals ${clds_reals_c_files} ${clds_reals_h_files})
```
Comment (amar-sagare) [inline tests/reals/CMakeLists.txt:28]: >clds_reals [](start = 12, length = 10)

aren't the other ones called `reals_module_name` ? at least that's what i used for bs2sf : `reals_bs2sf`

File: tests/reals/real_clds_singly_linked_list.h [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
 #define REAL_CLDS_SINGLY_LINKED_LIST_H
 
-#include "macro_utils.h"
+#include "azure_macro_utils/macro_utils.h"
 #include "clds/clds_singly_linked_list.h"
 
 #define R2(X) REGISTER_GLOBAL_MOCK_HOOK(X, real_##X);
 
 #define REGISTER_CLDS_SINGLY_LINKED_LIST_GLOBAL_MOCK_HOOKS() \
-    FOR_EACH_1(R2, \
```
Comment (amar-sagare) [inline tests/reals/real_clds_singly_linked_list.h:12]: >FOR_EACH_1 [](start = 4, length = 10)

interesting. How did this macro work before? #Closed

**Review by amar-sagare — APPROVED** [review]
:shipit:

File: tests/reals/real_clds_singly_linked_list.h (hunk 2 above)
Comment (amar-sagare) [reply tests/reals/real_clds_singly_linked_list.h:12]: Also, if macro_utls  is a local copy, is it going to be deleted?

---
In reply to: [370396422](https://github.com/dcristoloveanu/clds/pull/39#discussion_r370396422) [](ancestors = 370396422)

Comment (dcristoloveanu) [reply tests/reals/real_clds_singly_linked_list.h:12]: Well, this real was not used. And at some point there was a local macro_utils.h I believe.

File: tests/reals/CMakeLists.txt (hunk 1 above)
Comment (dcristoloveanu) [reply tests/reals/CMakeLists.txt:28]: I'll rename reals_bs2sf to bs2sf_reals when I pull them into the main repo. I don't quite care about the name, but it's in c util like this and I feel its more to rename the one in c util.

**Pull request merged**